<a href="https://colab.research.google.com/github/ivywanjage/cdav/blob/main/statistics.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [54]:
import pandas as pd

In [55]:
st_df=pd.read_csv("/content/student_performance.csv")

In [56]:
st_df.head()


,StudentID,Name,Gender,AttendanceRate,StudyHoursPerWeek,PreviousGrade,ExtracurricularActivities,ParentalSupport,FinalGrade
0,1,John,Male,85,15,78,1,High,80
1,2,Sarah,Female,90,20,85,2,Medium,87
2,3,Alex,Male,78,10,65,0,Low,68
3,4,Michael,Male,92,25,90,3,High,92
4,5,Emma,Female,88,18,82,2,Medium,85


In [57]:
st_df.columns

Index(['StudentID', 'Name', 'Gender', 'AttendanceRate', 'StudyHoursPerWeek',
       'PreviousGrade', 'ExtracurricularActivities', 'ParentalSupport',
       'FinalGrade'],
      dtype='object')

In [58]:
grade=st_df['FinalGrade']
grade

,FinalGrade
0,80
1,87
2,68
3,92
4,85
5,90
6,62
7,78
8,72
9,88


In [78]:
attendance=st_df["AttendanceRate"]
attendance

,AttendanceRate
0,85
1,90
2,78
3,92
4,88
5,95
6,70
7,85
8,82
9,91


In [79]:
attendance.mean()

np.float64(85.6)

In [80]:
attendance.std()

7.441624673267107

In [59]:
print("mean:",grade.mean())

mean: 80.2


In [60]:
print(f"median:{grade.median()}")

median:82.5


In [61]:
grade.mode()#no mode,it returns all the values

,FinalGrade
0,62
1,68
2,72
3,78
4,80
5,85
6,87
7,88
8,90
9,92


In [62]:
#quartiles
print("Q1: ",grade.quantile(0.25))
print("Q2: ",grade.quantile(0.50))
print("Q3: ",grade.quantile(0.75))

Q1:  73.5
Q2:  82.5
Q3:  87.75


q3--75% of the students scored less than or equal to 87.75

In [63]:
#deciles
print("D1: ",grade.quantile(0.10))
print("D2: ",grade.quantile(0.20))
print("D3: ",grade.quantile(0.30))

D1:  67.4
D2:  71.2
D3:  76.2


In [64]:
#percentile
print("P1: ",grade.quantile(0.30))

P1:  76.2


In [65]:
max=grade.max()
print("highesr grade is ",max)
min=grade.min()
print("lowest grade is ",min)

range=max-min
print("range is ",range)

Q3=grade.quantile(0.75)
Q1=grade.quantile(0.25)
IQR=Q3-Q1
print("IQR is ",IQR)

highesr grade is  92
lowest grade is  62
range is  30
IQR is  14.25


In [66]:
#variance/std deviation
var=grade.var()
std=grade.std()
var

101.95555555555555

In [67]:
std

10.09730437074943

In [68]:
grade.describe()

,FinalGrade
count,10.000000
mean,80.200000
std,10.097304
min,62.000000
25%,73.500000
50%,82.500000
75%,87.750000
max,92.000000


In [69]:
grade.describe(percentiles=[0.10,0.20,0.25])

,FinalGrade
count,10.000000
mean,80.200000
std,10.097304
min,62.000000
10%,67.400000
20%,71.200000
25%,73.500000
50%,82.500000
max,92.000000
